### PR TITLE
add fee action

### PR DESCRIPTION
**Pull request by me-vadim**
System contract 'fee' action

## Change Description
'fee' action will be used to fill reward pool when actions that require fee are executed

## Consensus Changes
Add new source to fill block_producers reward pool.

### CODE REVIEW

File: contracts/contracts/eosio.system/src/producer_pay.cpp [hunk 1]
```diff
@@ -195,4 +195,29 @@ namespace eosiosystem {
...
+      require_auth( payer );
+      auto amount = quantity.amount;
+      auto to_per_block_pay = amount * 0.2; //TODO: move 0.2 to constants section
+      auto to_per_vote_pay  = amount * 0.7; //TODO: move 0.7 to constants section
+      auto to_remme         = amount - (to_per_block_pay + to_per_vote_pay);
+      if( amount > 0 ) {
+         {
+            token::transfer_action transfer_act{ token_account, { {_self, active_permission} } };
+            if( to_remme > 0 ) {
+               transfer_act.send( _self, saving_account, asset(to_remme, core_symbol()), "remme fee" ); //TODO: change saving_account => remme_account
```
Comment (roman-tik) [inline contracts/contracts/eosio.system/src/producer_pay.cpp]: That is fine to keep the "saving" account name. Please do not use the term "fee"

File: contracts/contracts/eosio.system/src/producer_pay.cpp [hunk 2]
```diff
@@ -195,4 +195,29 @@ namespace eosiosystem {
       }
    }
 
+   void system_contract::fee( const name& payer, const asset& quantity ) {
+      require_auth( payer );
+      auto amount = quantity.amount;
```
Comment (mark-tsyrulnyk) [inline contracts/contracts/eosio.system/src/producer_pay.cpp]: @me-vadim please use `const` where possible

File: contracts/contracts/eosio.system/src/producer_pay.cpp [hunk 3]
```diff
@@ -195,4 +195,29 @@ namespace eosiosystem {
...
+      auto to_per_block_pay = amount * 0.2; //TODO: move 0.2 to constants section
+      auto to_per_vote_pay  = amount * 0.7; //TODO: move 0.7 to constants section
+      auto to_remme         = amount - (to_per_block_pay + to_per_vote_pay);
+      if( amount > 0 ) {
+         {
```
Comment (mark-tsyrulnyk) [inline contracts/contracts/eosio.system/src/producer_pay.cpp]: @me-vadim do we need this extra block?

File: contracts/contracts/eosio.system/src/producer_pay.cpp (hunk 1 above)
Comment (me-vadim) [reply contracts/contracts/eosio.system/src/producer_pay.cpp]: @roman-tik  What term should we use instead of "fee"?

**Review by mark-tsyrulnyk — APPROVED** [review]

**Review by olexiybuyanskyy — APPROVED** [review]

**Review by olexiybuyanskyy — APPROVED** [review]

**Pull request merged**